OpenLIFULib/OpenLIFULib/algorithm_input_widget.py

```diff
@@ -20,97 +20,99 @@ def indicate_no_options(self):
+        """
 
         layout = qt.QFormLayout(self)
         self.setLayout(layout)
 
-        self.protocol_input = AlgorithmInput("Protocol", qt.QComboBox(self))
-        self.transducer_input = AlgorithmInput("Transducer", qt.QComboBox(self))
-        self.volume_input = AlgorithmInput("Volume", qt.QComboBox(self))
-        self.target_input = AlgorithmInput("Target", qt.QComboBox(self))
-
-        self.inputs : List[AlgorithmInput] = [
-            self.protocol_input,
-            self.transducer_input,
-            self.volume_input,
-            self.target_input,
-        ]
-
-        for input in self.inputs:
+        self.inputs_dict = {}
+        for input_name in algorithm_input_names:
+            if input_name not in ["Protocol", "Transducer", "Volume", "Target", "Photoscan"]:
+                raise ValueError("Invalid algorithm input specified.")
+            else:
+                self.inputs_dict[input_name] = AlgorithmInput(input_name, qt.QComboBox(self))
+                
+        for input in self.inputs_dict.values():
             layout.addRow(f"{input.name}:", input.combo_box)
 
     def add_protocol_to_combobox(self, protocol : SlicerOpenLIFUProtocol) -> None:
-        self.protocol_input.combo_box.addItem("{} (ID: {})".format(protocol.protocol.name,protocol.protocol.id), protocol)
+        self.inputs_dict["Protocol"].combo_box.addItem("{} (ID: {})".format(protocol.protocol.name,protocol.protocol.id), protocol)
 
     def add_transducer_to_combobox(self, transducer : SlicerOpenLIFUTransducer) -> None:
         transducer_openlifu = transducer.transducer.transducer
-        self.transducer_input.combo_box.addItem("{} (ID: {})".format(transducer_openlifu.name,transducer_openlifu.id), transducer)
+        self.inputs_dict["Transducer"].combo_box.addItem("{} (ID: {})".format(transducer_openlifu.name,transducer_openlifu.id), transducer)
 
     def add_volume_to_combobox(self, volume_node : vtkMRMLScalarVolumeNode) -> None:
-        self.volume_input.combo_box.addItem("{} (ID: {})".format(volume_node.GetName(),volume_node.GetID()), volume_node)
+        self.inputs_dict["Volume"].combo_box.addItem("{} (ID: {})".format(volume_node.GetName(),volume_node.GetID()), volume_node)
+
+    def add_photoscan_to_combobox(self, model_node : vtkMRMLModelNode) -> None:
+        self.inputs_dict["Photoscan"].combo_box.addItem("{} (ID: {})".format(model_node.GetName(),model_node.GetID()), model_node)
 
     def set_session_related_combobox_tooltip(self, text:str):
-        """Set tooltip on the transducer and volume comboboxes."""
-        for input in [
-            self.protocol_input,
-            self.transducer_input,
-            self.volume_input,
-        ]:
-            input.combo_box.setToolTip(text)
+        """Set tooltip on the transducer, protocol and volume comboboxes."""
+
+        for input in ["Protocol", "Transducer", "Volume"]:
+            if input in self.inputs_dict.keys():
```

Comment:
```suggestion
            if input in self.inputs_dict:
```

Minor style comment for anywhere you have `x in d.keys()` : in python this is equivalent to `x in d` which looks a bit nicer. No need to change it just nice to know

Reply:
Thanks, didn't realize that!